core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/UserAuthMisoRequestManager.java

```diff
@@ -2615,4 +2615,51 @@ public Long countRuns() throws IOException {
   public Long countRunsBySearch(String querystr) throws IOException {
     return backingManager.countRunsBySearch(querystr);
   }
+
+  @Override
+  public Run getLatestRunBySequencerPartitionContainerId(Long containerId) throws IOException {
+    Run o = backingManager.getLatestRunBySequencerPartitionContainerId(containerId);
+    if (readCheck(o))
+      return o;
+    else
+      throw new IOException("User " + getCurrentUser().getFullName() + " cannot read Run " + o.getId());
+  }
+
+  @Override
+  public Long countContainers() throws IOException {
+    return backingManager.countContainers();
+  }
+
+  @Override
+  public List<SequencerPartitionContainer<SequencerPoolPartition>> getContainersByPageSizeSearch(int offset, int limit, String querystr,
+      String sortDir, String sortCol) throws IOException {
+    User user = getCurrentUser();
+    List<SequencerPartitionContainer<SequencerPoolPartition>> accessibles = new ArrayList<>();
+    for (SequencerPartitionContainer<SequencerPoolPartition> spc : backingManager
+        .getContainersByPageSizeSearch(offset, limit, querystr, sortDir, sortCol)) {
+      if (spc.userCanRead(user)) {
+        accessibles.add(spc);
+      }
+    }
+    return accessibles;
+  }
+
+  @Override
+  public List<SequencerPartitionContainer<SequencerPoolPartition>> getContainersByPageAndSize(int offset, int limit, String sortDir,
+      String sortCol) throws IOException {
+    User user = getCurrentUser();
+    List<SequencerPartitionContainer<SequencerPoolPartition>> accessibles = new ArrayList<>();
+    for (SequencerPartitionContainer<SequencerPoolPartition> spc : backingManager
+        .getContainersByPageAndSize(offset, limit, sortDir, sortCol)) {
+      if (spc.userCanRead(user)) {
+        accessibles.add(spc);
+      }
+    }
+    return accessibles;
+  }
+
+  @Override
+  public Long countContainersBySearch(String querystr) throws IOException {
```

Comment:
Let's leave it as is, as I think most of our internal users have access to most items. We can re-examine it if it ends up glitching due to users lacking permissions on a large number of items.